apps/api/test/08-report-8.e2e-spec.ts

```diff
@@ -0,0 +1,131 @@
+import { INestApplication } from '@nestjs/common';
+import { TestingModule, Test } from '@nestjs/testing';
+import request from 'supertest';
+import dayjs from 'dayjs';
+
+import { AppModule } from 'src/app.module';
+import { AuthGuard } from 'src/auth/auth.guard';
+import { URLS } from './constants';
+import { canActivate } from './override-guard';
+import { IENHaPcn } from 'src/applicant/entity/ienhapcn.entity';
+import { addApplicant, addMilestone, getHAs, hire, setApp } from './report-request-util';
+import { getApplicant, getStatusId } from './report-util';
+import { STATUS } from '@ien/common';
+
+describe('Report 8 - Registrants Working in BC', () => {
+  let app: INestApplication;
+  let applicantId: string;
+  let HA: IENHaPcn[] = [];
+
+  const lastYear = dayjs().year() - 1;
+  const fiscalDate = dayjs(`${lastYear}-06-06`).format('YYYY-MM-DD');
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideGuard(AuthGuard)
+      .useValue({ canActivate })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    setApp(app);
+    HA = await getHAs();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  const getReport8 = async () => {
+    const { body } = await request(app.getHttpServer()).get(URLS.REPORT8);
+    return body;
+  };
+
+  it('Add status to fiscal year', async () => {
+    const before = await getReport8();
+
+    const applicant = getApplicant();
+    applicant.registration_date = '2022-06-01';
```

Comment:
Will this break in a year? Would it be possible to use dayjs to have this be 6 months in the past or something like that?

Reply:
It will not.  That's just the applicants registration date, report 8 is looking for the date of `Received Work Permit (Arrival in Canada)`. A few lines down the start_date is '2021-01-01' so moving forward it should never be included in the report

Reply:
It will not.  That's just the applicants registration date, report 8 is looking for the date of `Received Work Permit (Arrival in Canada)`. A few lines down the start_date is dayjs(`${lastYear}-06-06`).format('YYYY-MM-DD');so moving forward it should never fail